flink-table-store-core/src/main/java/org/apache/flink/table/store/file/utils/FileStorePathFactory.java

```diff
@@ -61,20 +61,24 @@ public FileStorePathFactory(Path root, RowType partitionType, String defaultPart
         this.root = root;
         this.uuid = UUID.randomUUID().toString();
 
-        String[] partitionColumns = partitionType.getFieldNames().toArray(new String[0]);
-        this.partitionComputer =
-                new RowDataPartitionComputer(
-                        defaultPartValue,
-                        partitionColumns,
-                        partitionType.getFields().stream()
-                                .map(f -> LogicalTypeDataTypeConverter.toDataType(f.getType()))
-                                .toArray(DataType[]::new),
-                        partitionColumns);
+        this.partitionComputer = getPartitionComputer(partitionType, defaultPartValue);
 
         this.manifestFileCount = new AtomicInteger(0);
         this.manifestListCount = new AtomicInteger(0);
     }
 
+    public static RowDataPartitionComputer getPartitionComputer(
```

Comment:
Just for test?

Reply:
> Just for test?

Maybe for logging too. Currently, the log does not reveal the readable partition

Reply:
Add a `@VisibleForTesting`?